sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -227,20 +227,37 @@ trait SpillableHandle extends StoreHandle with Logging {
   private[spill] def doClose(): Unit
 
   /**
-   * When the current handle is actually doing spilling, it might encounter exceptions, e.g.
-   * InterruptedException. This method is used to catch those exceptions and reset spilling state
-   * to false
+   * Executes a spill operation while handling exceptions and updating metrics.
+   * If an exception occurs during spilling (e.g. InterruptedException), this method
+   * catches it and resets the spilling state to false. Metrics are tracked separately
+   * for disk spills vs host spills.
    */
-  def inCaseSpillingFailed(block: () => Long): Long = {
-    try {
-      block()
-    } catch {
-      case e: Throwable =>
-        logWarning("Failed to spill SpillableColumnarBatchHandle", e)
-        synchronized {
-          spilling = false
-        }
-        throw e
+  def executeSpill(spillToDisk: Boolean)(block: () => Long): Long = {
```

Comment:
Lets add to the function comment, a sentence about  `block` returning bytes spilled.

Reply:
Refined the comment